### PR TITLE
4475 Fix for aggregation copy 

### LINKED ISSUE #4475: Aggregations are not created as part of resource copy and version
**Description of the bug**
If a resource has folder based aggregations (fileset, model instance or model program) and the folder representing the aggregation is empty, then copying or creating a new version of the resource creates a resource that doesn't contain the aggregation. However, the empty folder can be found in the new resource.

Steps to reproduce the bug:
1. Create a resource
2. Create a file set aggregation after you create a folder and uploading a file to that folder. Add some aggregation level metadata.
3. Delete the file you uploaded
4. Create a copy of the resource. 
5. If you check the copied resource you should see the folder but the fileset aggregation should be missing.
6. Create a version of the original resource.
7. If you check the new version of the resource you should see the folder but the fileset aggregation should be missing.

**Expected behavior**
When a resource with aggregation is copied or a new version is created, the new resource should contain the aggregation.

**Additional information**
Add any other context about the problem here such as screenshots, software versions, and browser information.

### PR DESCRIPTION
**Pull request by pkdash**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create resource.
2. Create a folder and upload a file to that folder. Set the folder to file set aggregation.
3. Delete the file that you uploaded. The leaves the file set aggregation with no files.
4. Create another folder and upload a file to that folder. Set the folder to Model Instance aggregation. 
5. Delete the file you uploaded in step 4 above. This leaves the Model Instance aggregation with no files.
6. Repeat step 4 and 5  for Model Program aggregation.
7. Make a copy of the resource. Verify the copied resource has one file set aggregation, one model instance aggregation and one model program aggregation.

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4482](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4482)

File: pytest/functional/hs_file_types/test_model_aggregation.py [hunk 1]
```diff
@@ -876,6 +876,80 @@ def test_resource_copy(composite_resource, mock_irods):
...
     assert copy_mi_aggr.files.count() == 1
     assert copy_mp_aggr.files.count() == 1
+    assert copy_mi_aggr.metadata.executed_by == copy_mp_aggr
+
+
+@pytest.mark.django_db(transaction=True)
+def test_resource_copy_with_aggregations_with_no_files(composite_resource, mock_irods):
+    """Test copying a resource that has both mi and mp folder based aggregations (aggregations have no files)"""
+
+    res, user = composite_resource
```
[the remark this answers is not in the document]
Comment (pkdash) [reply pytest/functional/hs_file_types/test_model_aggregation.py:886]: `compsite_resource` is a pytest fixture function that returns 2 values (a resource and a user)

File: hs_composite_resource/tests/test_composite_resource.py [hunk 2]
```diff
@@ -2018,6 +2018,41 @@ def test_copy_resource_with_file_set_aggregation_3(self):
...
 
+    def test_copy_resource_with_file_set_aggregation_4(self):
+        """Here we are testing that we can create a copy of a composite resource that contains one
+        file set aggregation where the file set aggregation has no files"""
+
+        self.create_composite_resource()
+        self.assertEqual(CompositeResource.objects.count(), 1)
+        new_folder = 'fileset_folder'
+        ResourceFile.create_folder(self.composite_resource, new_folder)
+        # add the the txt file to the resource at the above folder
```
[the remark this answers is not in the document]
Comment (pkdash) [reply hs_composite_resource/tests/test_composite_resource.py]: fixed

**Comment by pkdash** [comment]
@hyi See if I have addressed your comments.

**Review by hyi — APPROVED** [review]

**Pull request merged**